mailbox/api/pom.xml

```diff
@@ -32,6 +32,10 @@
     <name>Apache James :: Mailbox :: API</name>
 
     <dependencies>
+        <dependency>
+            <groupId>${james.groupId}</groupId>
+            <artifactId>blob-api</artifactId>
```

Comment:
By removing the BlobId abstraction, we tighnen the coupling between blobs and the fact that we use a hash as their primary access key.
It may make some things simpler but I think that we loose a lot of flexibility.
It also add more links between components. Why would mailbox-api requires blob-api?

Reply:
The backup need the BlobStore from the blob-api.

Reply:
But I will remove the refactoring considering the BlobId.